json-path/src/main/java/com/jayway/jsonpath/Predicate.java

```diff
@@ -23,6 +24,8 @@
 
     boolean apply(PredicateContext ctx);
 
+    boolean check(TokenStack stack, int idx);
```

Comment:
FYI, this is a pet project of mine that I work with on my free time, there is no one else. I appreciate all contributions but I also need to feel comfortable supporting them in the future. I have have made some breaking changes to the API over the years and streaming support is defiantly a feature that I would like.   

Sorry about the slow pace and lack of comments, I hope to have more time for this project soon.